submissions/dikun/frogger-game/app.js

```diff
@@ -0,0 +1,123 @@
+"use srtict";
+
+const cell = {
+	width: 100,
+	height: 85
+};
+
+const gameField = {
+	top: 0,
+	right: cell.width * 5,
+	bottom: cell.height * 5,
+	left: 0
+};
+
+const playerStart = {
+	x: 202,
+	y: 400
+};
+
+const enemySpeed = {
+	min: 90,
+	max: 250 
+}; 
+
+const sprite = {
+	enemy: 'images/enemy-bug.png',
+	player: 'images/char-boy.png'
+};
+
+const speed = () => enemySpeed.min + Math.floor(Math.random() * enemySpeed.max);
+const crash = 70;
+const border = 20;
+
+const Enemy = function (x, y, speed, player) {
+	this.x = x;
+	this.y = y;
+	this.speed = speed();
+	this.player = player;
+	this.sprite = sprite.enemy;	
+
+	this.checkColission = function () {
+		if(this.player.x + crash > this.x &&
+			this.player.x < this.x + crash &&
+			this.player.y + crash > this.y &&
+			this.player.y < this.y + crash) {
+			this.player.x = playerStart.x;
+			this.player.y = playerStart.y;
+		}
+	};
```

Comment:
This approach defines a **copy** of function per every instance of class.
We do not want this. One of the crucial points of prototypical inheritance model is to have a single copy of method per class, stored in its prototype object.
Check the original project code for a pattern to follow.